makefiles/tools/edbg.inc.mk

```diff
@@ -1,9 +1,15 @@
 include $(RIOTMAKE)/tools/edbg-devices.inc.mk
 
+# Edbg use a bin file for flashing
+FLASHFILE ?= $(BINFILE)
+
 RIOT_EDBG = $(RIOTTOOLS)/edbg/edbg
 EDBG ?= $(RIOT_EDBG)
-FLASHER ?= $(EDBG)
-FLASHFILE ?= $(BINFILE)
+ifneq (0,$(shell command -v $(EDBG) 2>&1 > /dev/null ; echo $$?))
+  FLASHDEPS += $(RIOT_EDBG)
+  EDBG := $(RIOT_EDBG)
+endif
```

Comment:
Why did you need to add this? Wouldn't EDBG be part of the dependencies enough?

Reply:
Good question ! In #15970, for some reason, I always had the message that edbg wasn't found, and the tool Makefile was always triggered. This change fixed it but I agree it's not super nice.

Let me try with the previous logic in this PR.

Reply:
Ok so I tried and the problem with the previous logic is that it checks that the FLASHER variable is the same as EDBG. But with PR, FLASHER doesn't correspond anymore to the edbg binary but to the wrapper script.

In the end, it turns out that using the simple:

```mk
EDBG ?= $(RIOTTOOLS)/edbg/edbg
FLASHDEPS += $(EDBG)
```

works the same and is simpler:

- without edbg available in RIOT, it's correctly built
 
```
$ make -C dist/tools/edbg clean  --no-print-directory 
rm -f /work/riot/RIOT/dist/tools/edbg/edbg
$ make BOARD=samr21-xpro -C examples/hello-world flash-only --no-print-directory 
[INFO] edbg binary not found - building it from source now
CC= CFLAGS= make -C /work/riot/RIOT/dist/tools/edbg
env -i PATH="/home/aabadie/.local/bin:/work/tools/mips-mti-elf/2016.05-03/bin:/work/tools/avr8-gnu-toolchain-linux_x86_64/bin:/work/tools/riscv-none-gcc/xpack-riscv-none-embed-gcc-10.1.0-1.1/bin:/work/tools/gcc-arm-none-eabi-9-2019-q4-major/bin/:/home/aabadie/.cargo/bin:/home/aabadie/.cargo/bin:/home/aabadie/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/work/tools/packer" TERM="xterm-256color" "make" -C "/work/riot/RIOT/dist/tools/edbg/bin"
make: Entering directory '/work/riot/RIOT/dist/tools/edbg/bin'
gcc -W -Wall -Wextra -O2 -std=gnu11 dap.c edbg.c target.c target_atmel_cm0p.c target_atmel_cm3.c target_atmel_cm4.c target_atmel_cm7.c target_atmel_cm4v2.c target_mchp_cm23.c target_st_stm32g0.c target_gd_gd32f4xx.c  dbg_lin.c -ludev -o edbg
make: Leaving directory '/work/riot/RIOT/dist/tools/edbg/bin'
mv /work/riot/RIOT/dist/tools/edbg/bin/edbg .
[INFO] edbg binary successfully built!
/work/riot/RIOT/dist/tools/edbg/edbg.sh flash /work/riot/RIOT/examples/hello-world/bin/samr21-xpro/hello-world.bin
### Flashing Target ###
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800009840 01.1A.00FB (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev C)
Verification....................................... done.
Done flashing
```

- if edbg is available in RIOT, it's not re-built
 
```
$ make BOARD=samr21-xpro -C examples/hello-world flash-only --no-print-directory 
/work/riot/RIOT/dist/tools/edbg/edbg.sh flash /work/riot/RIOT/examples/hello-world/bin/samr21-xpro/hello-world.bin
### Flashing Target ###
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800009840 01.1A.00FB (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev C)
Verification....................................... done.
Done flashin
```